src/template.rs

```diff
@@ -148,29 +151,96 @@ pub(crate) fn walk_dir(
+                    }
+                }
+            }
         }
     }
+    if files_with_errors.len() > 0 {
+        let warn = construct_substitution_warning(files_with_errors);
+        println!("{}", warn);
+    }
 
     pbar.finish_and_clear();
     Ok(())
 }
+
+fn parse_and_replace(
+    context: &Object,
+    filename: &Path,
+    parsed_file: Template,
+) -> Result<(), anyhow::Error> {
+    let new_content = parse_file(context, filename, &parsed_file);
+    fs::write(filename, new_content).with_context(|| {
+        format!(
+            "{} {} `{}`",
+            emoji::ERROR,
+            style("Error writing").bold().red(),
+            style(filename.display()).bold()
+        )
+    })?;
+
+    Ok(())
+}
+
+fn parse_file(context: &Object, filename: &Path, parsed_file: &Template) -> String {
+    match parsed_file.render(context) {
+        Ok(content) => content,
+        Err(e) => {
+            let msg = e.to_string();
+            if msg.contains("requested variable") {
+                // so, we miss a variable that is present in the file to render
+                let requested_var =
+                    regex::Regex::new(r"(?P<p>.*requested\svariable=)(?P<v>.*)").unwrap();
+                let captures = requested_var.captures(msg.as_str()).unwrap();
+
+                if let Some(Some(req_var)) = captures.iter().last() {
+                    let missing_variable = req_var.as_str().to_string();
+                    // try again with this variable added to the context
+                    let mut context = context.clone();
+                    context.insert(missing_variable.into(), Value::scalar("".to_string()));
+                    // now let's parse again to see if we have all variables declared now
+                    return parse_file(&context, filename, parsed_file);
+                }
+            }
+            eprintln!(
+                "{} {} `{}`",
+                emoji::ERROR,
+                style("Error replacing placeholders, file got only copied.")
+                    .bold()
+                    .red(),
+                style(filename.display()).bold()
+            );
+            // fallback: take the file as it is, without substitutions
+            fs::read_to_string(filename).unwrap()
+        }
+    }
+}
+
+fn construct_substitution_warning(files_with_errors: Vec<(String, liquid_core::Error)>) -> String {
+    let mut msg = format!(
+        "\n{} {}",
+        emoji::WARN,
+        style("Substitution skipped, found invalid syntax in\n")
+            .bold()
+            .red(),
+    );
+    for file_error in files_with_errors {
+        msg.push_str("\t");
+        msg.push_str(&file_error.0);
+        msg.push_str("\n");
+    }
+    msg.push_str("\n");
+    let info = format!("{}", style("Consider adding these files to a `cargo-generate.toml` in the template repo to skip substitution on these files.\n").bold());
+    msg.push_str(&info);
```

Comment:
This requires `use std::fmt::Write` at the top of the file, but:
```suggestion
    writeln!(msg, "{}", style("Consider adding these files to a `cargo-generate.toml` in the template repo to skip substitution on these files.").bold()).unwrap();
```

Reply:
unfortunately this won't compile:

```
error[E0599]: no method named `write_fmt` found for struct `std::string::String` in the current scope
   --> src/template.rs:236:5
    |
236 |     writeln!(msg, "{}", style("Consider adding these files to a `cargo-generate.toml` in the template repo to skip substitution on these files.").bold()).un...
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ method not found in `std::string::String`
    |
    = help: items from traits can only be used if the trait is in scope
    = note: the following trait is implemented but not in scope; perhaps add a `use` for it:
            `use std::fmt::Write;`
    = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

error: could not compile `cargo-generate`.
```

Reply:
@sassman I think you forgot the `use std::fmt::Write` as Veetaha stated before :)

Reply:
sorry my bad. fixed.